tombstones_test.go

```diff
@@ -0,0 +1,333 @@
+		},
+		{
+			exist: intervals{{5, 10}, {12, 20}, {25, 30}},
+			new:   interval{11, 14},
+			exp:   intervals{{5, 20}, {25, 30}},
+		},
+	}
+
+	for _, c := range cases {
+
+		require.Equal(t, c.exp, c.exist.add(c.new))
+	}
+	return
+}
+
+func TestTombstoneReadersSeek(t *testing.T) {
+	// This is assuming that the listPostings is perfect.
+	table := struct {
+		m map[uint32]intervals
+
+		cases []uint32
+	}{
+		m: map[uint32]intervals{
+			2:    intervals{{1, 2}},
+			3:    intervals{{1, 4}, {5, 6}},
+			4:    intervals{{10, 15}, {16, 20}},
+			5:    intervals{{1, 4}, {5, 6}},
+			50:   intervals{{10, 20}, {35, 50}},
+			600:  intervals{{100, 2000}},
+			1000: intervals{},
+			1500: intervals{{10000, 500000}},
+			1600: intervals{{1, 2}, {3, 4}, {4, 5}, {6, 7}},
+		},
+
+		cases: []uint32{1, 10, 20, 40, 30, 20, 50, 599, 601, 1000, 1600, 1601, 2000},
+	}
+
+	testFunc := func(t *testing.T, tr TombstoneReader) {
+		for _, ref := range table.cases {
+			// Create the listPostings.
+			refs := make([]uint32, 0, len(table.m))
+			for k := range table.m {
+				refs = append(refs, k)
+			}
+			sort.Sort(uint32slice(refs))
+			pr := newListPostings(refs)
+
+			// Compare both.
+			trc := tr.Copy()
+			require.Equal(t, pr.Seek(ref), trc.Seek(ref))
+			if pr.Seek(ref) {
+				require.Equal(t, pr.At(), trc.At().ref)
+				require.Equal(t, table.m[pr.At()], trc.At().intervals)
+			}
+
+			for pr.Next() {
+				require.True(t, trc.Next())
+				require.Equal(t, pr.At(), trc.At().ref)
+				require.Equal(t, table.m[pr.At()], trc.At().intervals)
+			}
+
+			require.False(t, trc.Next())
+			require.NoError(t, pr.Err())
+			require.NoError(t, tr.Err())
+		}
+	}
+
+	t.Run("tombstoneReader", func(t *testing.T) {
+		tmpdir, _ := ioutil.TempDir("", "test")
+		defer os.RemoveAll(tmpdir)
+
+		mtr := newMapTombstoneReader(table.m)
+		writeTombstoneFile(tmpdir, mtr)
+		tr, err := readTombstoneFile(tmpdir)
+		require.NoError(t, err)
+
+		testFunc(t, tr)
+		return
+	})
+	t.Run("mapTombstoneReader", func(t *testing.T) {
+		mtr := newMapTombstoneReader(table.m)
+
+		testFunc(t, mtr)
+		return
+	})
+	t.Run("simpleTombstoneReader", func(t *testing.T) {
+		dranges := intervals{{1, 2}, {3, 4}, {5, 6}}
+
+		for _, ref := range table.cases {
+			// Create the listPostings.
+			refs := make([]uint32, 0, len(table.m))
+			for k := range table.m {
+				refs = append(refs, k)
+			}
+			sort.Sort(uint32slice(refs))
+			pr := newListPostings(refs[:])
+			tr := newSimpleTombstoneReader(refs[:], dranges)
+
+			// Compare both.
+			trc := tr.Copy()
+			require.Equal(t, pr.Seek(ref), trc.Seek(ref))
+			if pr.Seek(ref) {
+				require.Equal(t, pr.At(), trc.At().ref)
+				require.Equal(t, dranges, tr.At().intervals)
+			}
+			for pr.Next() {
+				require.True(t, trc.Next())
+				require.Equal(t, pr.At(), trc.At().ref, "refs")
+				require.Equal(t, dranges, trc.At().intervals)
+			}
+
+			require.False(t, trc.Next())
+			require.NoError(t, pr.Err())
+			require.NoError(t, tr.Err())
+		}
+		return
```

Comment:
Yes, in testFunc we are testing against the map, but here we are testing against a static range.